migrating_to_2.0/recipes.rst

```diff
@@ -205,6 +205,16 @@ are valid.
             if self.info.settings.os == "Windows":
                 raise ConanInvalidConfiguration("This package is not compatible with Windows")
 
+.. note::
+
+    For recipes where settings are cleared, using ``self.settings`` is still valid. For example,
+    this applies to header only recipes that check for a specific ``self.settings.cppstd`` like:
+
+    .. code-block:: python
+
+        def validate(self):
```

Comment:
```suggestion
        def package_id(self):
            self.info.clear()

        def validate(self):
```
Make it more explicit according to the note. Anything removing compiler from the package ID should follow the same path of course.